workflow-templates/appstore-build-publish.yml

```diff
@@ -68,24 +71,42 @@ jobs:
           php-version: ${{ env.PHP_VERSION }}
           coverage: none
 
+      - name: Install composer dependencies
+        # Continue if no composer.json
+        continue-on-error: true
+        run: |
+          cd ${{ env.APP_NAME }}
+          composer install --no-dev
+
       - name: Build ${{ env.APP_NAME }}
+        # Skip if no package.json
         if: ${{ steps.versions.outputs.nodeVersion }}
         run: |
           cd ${{ env.APP_NAME }}
           npm ci
           npm run build
 
       - name: Package ${{ env.APP_NAME }} ${{ env.APP_VERSION }}
+        # Try krankerl, fallback to makefile
         run: |
           cd ${{ env.APP_NAME }}
           krankerl package || make appstore
 
       - name: Checkout server ${{ fromJSON(steps.appinfo.outputs.result).nextcloud.min-version }}
+        continue-on-error: true
+        id: server-checkout
         run: |
           NCVERSION=${{ fromJSON(steps.appinfo.outputs.result).nextcloud.min-version }}
           wget --quiet https://download.nextcloud.com/server/releases/latest-$NCVERSION.zip
           unzip latest-$NCVERSION.zip
 
+      - name: Checkout server master fallback
+        uses: actions/checkout@v2
+        if: ${{ steps.server-checkout.outcome != 'success' }}
```

Comment:
Maybe we can do something like this with the composer.json to check if deps are needed?

Reply:
Done